File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    arr1 = [1,2,3,4,5] * 2001
    arr2 = [2,1,2,3,2,4,2,5] * 1251
    arr3 = [3,3,1,1,2,2,4,4,5,5] * 1001
    
    arr1 = arr1[:len(answers)]
    arr2 = arr2[:len(answers)]
    arr3 = arr3[:len(answers)]
    
    cnt = [0, 0, 0]
    for i in range(len(answers)):
        if arr1[i] == answers[i]:
            cnt[0] += 1
        if arr2[i] == answers[i]:
            cnt[1] += 1
        if arr3[i] == answers[i]:
            cnt[2] += 1
            
    max_num = max(cnt)
    
    for i in range(len(cnt)):
        if max_num == cnt[i]:
            answer.append(i+1)
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    static int NUM = 3;
    
    public int[] solution(int[] answers) {
        int[] idx = new int[NUM];
        int[][] student = new int[][] { 
            {1, 2, 3, 4, 5},
            {2, 1, 2, 3, 2, 4, 2, 5},
            {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
        };
        int[] cnt = new int[NUM];
        
        for (int i = 0; i < answers.length; i++) {
            for (int j = 0; j < NUM; j++) {
                int index = i % student[j].length;
                if (answers[i] == student[j][index]) cnt[j]++;
            }
        }
        
        // cnt에서 최댓값 찾고, 최댓값인 index로 새 배열 만들기
        int max = Arrays.stream(cnt).max().getAsInt();
        ArrayList<Integer> answer = new ArrayList<>();
        for (int i = 0; i < NUM; i++) {
            if (cnt[i] == max) answer.add(i+1);
        }
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}